Clearer management of files when assigning typesetting

joemull opened the pull request:
This tackles a set of related issues with file management during typesetting assignments, especially in subsequent rounds of typesetting involving proofing files and galley correction.

- Closes #166 
- Closes #194 
- Closes #152 
- Closes #154
- Closes #140
- Closes #138
- Closes BirkbeckCTP/janeway#4314

Regarding #152, this PR does not allow editors to individually select proofreaders' comments, but it does allow them to individually select proofing files separately from comments, and upload their own proofing files. If they want to collate the comments together, they can do this more easily given the summary/details display of proofing comments.

Dependencies:

- BirkbeckCTP/janeway#4327
- BirkbeckCTP/janeway#4328

## Editor's main view of typesetting files

![image](https://github.com/user-attachments/assets/a5c030b7-903c-4262-88b1-d125e22ceb9b)

## Editor assigns a typesetter

![image](https://github.com/user-attachments/assets/91b51908-39d6-48b2-88f7-48c93d71585a)
![image](https://github.com/user-attachments/assets/083e93f5-5105-48ce-83f2-ddd36851a68a)
![image](https://github.com/user-attachments/assets/e282b3e4-0e88-4461-9309-6dd6d0c75c7f)
![image](https://github.com/user-attachments/assets/31b40ad3-882a-4da1-8013-3f8918555fbd)

## Typesetter views the assignment

![image](https://github.com/user-attachments/assets/691d9163-7952-49ef-b140-c11c0ab0614e)
![image](https://github.com/user-attachments/assets/4031470c-1c23-4e7f-b23d-a8402a4acf17)

ajrbyers commented on logic.py:
```diff
@@ -22,11 +23,13 @@
     :param file_objects: Boolean
     :return: a list of File type objects
     """
+    raise DeprecationWarning('Use get_typesetting_files instead.')
```
Just a note: this will mean a major version bump for the typesetting plugin.

joemull replied:
I only learned this recently but warnings don't cause an exception or terminate the program, so this is what we want to do some time *before* advancing the major version. Removing the function entirely is the change to make when we do a major version bump.

https://docs.python.org/3/library/warnings.html

ajrbyers replied:
> I only learned this recently but warnings don't cause an exception or terminate the program, so this is what we want to do some time _before_ advancing the major version. Removing the function entirely is the change to make when we do a major version bump.
> 
> https://docs.python.org/3/library/warnings.html

Ace!

ajrbyers commented on views.py:
```diff
@@ -278,6 +278,8 @@ def typesetting_upload_galley(request, article_id, assignment_id=None):
         )
 
     if 'prod' in request.POST:
```
This should just be removed.

ajrbyers reviewed (changes requested):
Thanks for this @joemull I think this will help editors out a lot with this step in the process.

- I've added two review comments
- I do have a concern around not presenting the typesetters, automatically, with figure files. Do you think editors will be okay with selecting them? I think journals like CG will consider this a step backwards as they often have large numbers of images that will need to be selected.
- For articles in typesetting we will need a data migration that will add the figures to the list of files for the typesetter otherwise they will lose access to them.

Let me know what you think about the above.

joemull commented:
>     * I do have a concern around not presenting the typesetters, automatically, with figure files. Do you think editors will be okay with selecting them? I think journals like CG will consider this a step backwards as they often have large numbers of images that will need to be selected.

I think we got into trouble with the usability of this interface by automatically controlling the visibility of things, so what if we make a button with which editors can select all figure files, if they want to? I think I can expand the select_all widget to support looking for value in a `data-` attribute, and we could  use the result of `file_type` there.

>     * For articles in typesetting we will need a data migration that will add the figures to the list of files for the typesetter otherwise they will lose access to them.

I can write a migration for this.

joemull commented:
Or we could add a setting that controls whether they are pre-selected.

Or if we think most people will want this, we could just preselect them always, without a setting. A few other things are now preselected, so this would not be unique behavior.

Thoughts @ajrbyers ?

joemull commented:
@ajrbyers I've gone ahead and pre-checked data and figure files.

ajrbyers reviewed (changes requested):
Generally this is really good.

1. When assigning a correction request the editor is presented with the option to select Galley files twice. They Typesetter will have access to the Galley regardless of whether the editor selects the file here.
<img width="824" alt="Screenshot 2024-08-14 at 15 35 59" src="https://github.com/user-attachments/assets/186e90d0-fe92-4f19-aa10-db8fba993695">

2. When logged in as a Typesetter and undertaking a correction task I was able to see the same Galley listed twice.
<img width="1367" alt="Screenshot 2024-08-14 at 15 40 18" src="https://github.com/user-attachments/assets/a1507ace-f89d-46d9-b18a-3b720157ff7c">

3. The article metadata has been moved to the foot of the Typesetter page. Would it be worth adding some sort of signpost? It could easily be missed.

joemull commented:
Thanks @ajrbyers.

> 1. When assigning a correction request the editor is presented with the option to select Galley files twice.

One part of the form asks "Which files should be used?" and the other asks "Which galleys need to be corrected?". The two are different things and it is important to be literal and clear in these interfaces, according to the user reports. The selected files control different lists on the typesetter's end, with different explanations (see below).

I prevent errors with the confirmable form:

https://github.com/BirkbeckCTP/typesetting/blob/07115db9480f3919d0bef0ec9ad614be38fbcc2e/forms.py#L92-L105

> They Typesetter will have access to the Galley regardless of whether the editor selects the file here.

That's how it used to work, but I changed it because that was one of the sources of confusion. The typesetter sees this text above the first list of files:

"The following files have been selected for use or reference when generating the typeset articles."

For "have been selected" to be true, it is important that all the files, and only the files, selected by the editor appear here. Otherwise the interface can contradict the editors' instructions, which is a source of confusion that users reported.

Granted, it is true that the galley files uploaded by the typesetter will always appear in a separate list. The text above that list says:

"Any galleys (typeset files) that were selected by the editor for correction, or that you originally uploaded, will appear here."

This list needs to stay separate so that the editors' selection is a meaningful piece of communication between editor and typesetter.

> 2. When logged in as a Typesetter and undertaking a correction task I was able to see the same Galley listed twice.

Thanks for catching that. I'll fix it.

> 3. The article metadata has been moved to the foot of the Typesetter page. Would it be worth adding some sort of signpost? It could easily be missed.

I've added a skip link. Doesn't look fantastic but it will do the trick for now:

![image](https://github.com/user-attachments/assets/5601bbdc-6191-47d4-98d0-aae18982015f)

ajrbyers reviewed (approved):
🥇
